cli-access-protected-api/src/main/java/com/contoso/cli/App.java

```diff
@@ -0,0 +1,81 @@
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.aad.msal4j.ClientCredentialFactory;
+import com.microsoft.aad.msal4j.ClientCredentialParameters;
+import com.microsoft.aad.msal4j.ConfidentialClientApplication;
+import com.microsoft.aad.msal4j.IAuthenticationResult;
+
+/**
+ * This application represents a confidential client application that is going
+ * to be calling Microsoft Graph as itself. This is a confidential client in
+ * that the application will need its own identity and security token to perform
+ * the work, and it is not doing this on behalf of an enterprise user.
+ * 
+ * An example of this might be a nightly job that runs a query against a
+ * database, and based on the results sends an email to multiple people in the
+ * organization.
+ */
+public class App {
+
+    public static void main(String[] args) throws IOException {
+
+        // All of your Azure AD configurations are stored in the application.properties
+        // file. Load those up for easy access.
+        final Properties properties = new Properties();
+        properties.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("application.properties"));
+
+        // This scenario uses a confidential client application MSAL client. The
+        // lifecycle of this client should be scoped to the application's lifecycle
+        // as it contains a token and metadata cache that will prevent unnecessary
+        // requests to Azure AD on subsequent usage.
+        final ConfidentialClientApplication msalClient = ConfidentialClientApplication.builder(
+                properties.getProperty("client-id"),
+                ClientCredentialFactory.createFromSecret(properties.getProperty("client-secret")))
+                .authority(properties.getProperty("authority"))
+                .build();
+
+        // Acquire a token from Azure AD for this client to access Microsoft Graph based
+        // on the permissions granted this application in its Azure AD App registration.
+        final ClientCredentialParameters parameters = ClientCredentialParameters
+                .builder(Collections.singleton("https://graph.microsoft.com/.default"))
+                .build();
+        final CompletableFuture<IAuthenticationResult> authResult = msalClient.acquireToken(parameters);
+        final IAuthenticationResult tokenResult = authResult.join();
+        final String graphAccessToken = tokenResult.accessToken();
+
+        // Access Microsoft Graph using the access token acquired above. This
+        // application is simply accessing its own App registration record in Azure AD.
+        final HttpRequest request = HttpRequest.newBuilder()
+                .uri(URI.create(
+                        "https://graph.microsoft.com/v1.0/applications/" + properties.getProperty("client-object-id")))
```

Comment:
Can you use `client-id`, instead?

Reply:
Unfortunately nope.  This is the same call we've done in other client credential grant samples as well, so we've been consistent in using them like this.  I really wish it would take client id.  Feature request? :)

Reply:
That's what I was seeing yesterday when I was poking around but I just figured I was missing something.

What about a GET of apps, filtering on app ID?